### ISSUE TITLE
SSL Pinning function

### ISSUE THREAD

**Issue by ammnt** [AdguardTeam/AdGuardHome#4806]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

### Description

Hello,

please add SSL pinning. Now I use NGINX as reverse proxy to solve this problem🙏🏼

Thank you.
Best regards😙

**Comment by centralhardware** [comment]
IMHO nginx is best place for ssl pining.

**Comment by 7ArxS2PrqAUF9Gom** [comment]
I agree. I think that there should either be 2 SSL certificates - one for GUI and one for domain resolution via DoH/DoT or an option to disable domain resolution via DoH/DoT, but preserve SSL encryption exclusively for GUI login. Routers use separate certificates for separate functions (server, bridge, guest captive portal, etc). It is actually less confusing and more secure that way.

I also think that for GUI login AGH team should create a key store that generates random self-signed SSL certificates during AGH setup.
